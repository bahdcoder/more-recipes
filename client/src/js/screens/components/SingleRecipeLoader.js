import React, { Component } from 'react';
import ContentLoader, { Rect, Circle } from 'react-content-loader';

export default class SingleRecipeLoader extends Component {
  render() {
    return (
      <div>
        <ContentLoader 
          speed={1} 
          primaryColor={'#f2c3a6'} 
          secondaryColor={'#eba57a'}
          width={680}
          height={700}>
        <Rect 
          height={400}
          width={680} />
      </ContentLoader>
      </div>
    );
  }
}
