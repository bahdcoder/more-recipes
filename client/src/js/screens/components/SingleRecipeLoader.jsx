import React from 'react';
import ContentLoader, { Rect } from 'react-content-loader';

const SingleRecipeLoader = () => ((
  <div>
    <ContentLoader
      speed={1}
      primaryColor="#f5d2bd"
      secondaryColor="#f9e0d2"
      width={680}
      height={700}
    >
      <Rect
        height={400}
        width={680}
      />
      <Rect y={420} radius={9} width={680} height={18} />
      <Rect y={460} radius={9} width={500} height={18} />
      <Rect y={500} radius={9} width={420} height={18} />
    </ContentLoader>
  </div>
));

export default SingleRecipeLoader;
