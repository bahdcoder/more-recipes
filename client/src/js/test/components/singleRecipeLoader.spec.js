/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import SingleRecipeLoader from '../../screens/components/SingleRecipeLoader';

import setupTests from '../setupTests';

setupTests();

describe('The SingleRecipeLoader component', () => {
  test('should mount without crashing', () => {
    const wrapper = shallow(<SingleRecipeLoader />);
    expect(wrapper).toMatchSnapshot();
  });
});
