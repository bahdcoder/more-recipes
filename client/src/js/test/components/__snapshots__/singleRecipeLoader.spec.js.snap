// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SingleRecipeLoader component should mount without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleRecipeLoader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ContentLoader
        height={700}
        primaryColor="#f5d2bd"
        secondaryColor="#f9e0d2"
        speed={1}
        type="facebook"
        width={680}
      >
        <Rect
          height={400}
          width={680}
        />
        <Rect
          height={18}
          radius={9}
          width={680}
          y={420}
        />
        <Rect
          height={18}
          radius={9}
          width={500}
          y={460}
        />
        <Rect
          height={18}
          radius={9}
          width={420}
          y={500}
        />
      </ContentLoader>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Rect
            height={400}
            width={680}
          />,
          <Rect
            height={18}
            radius={9}
            width={680}
            y={420}
          />,
          <Rect
            height={18}
            radius={9}
            width={500}
            y={460}
          />,
          <Rect
            height={18}
            radius={9}
            width={420}
            y={500}
          />,
        ],
        "height": 700,
        "primaryColor": "#f5d2bd",
        "secondaryColor": "#f9e0d2",
        "speed": 1,
        "type": "facebook",
        "width": 680,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "height": 400,
            "width": 680,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "height": 18,
            "radius": 9,
            "width": 680,
            "y": 420,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "height": 18,
            "radius": 9,
            "width": 500,
            "y": 460,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "height": 18,
            "radius": 9,
            "width": 420,
            "y": 500,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ContentLoader
          height={700}
          primaryColor="#f5d2bd"
          secondaryColor="#f9e0d2"
          speed={1}
          type="facebook"
          width={680}
        >
          <Rect
            height={400}
            width={680}
          />
          <Rect
            height={18}
            radius={9}
            width={680}
            y={420}
          />
          <Rect
            height={18}
            radius={9}
            width={500}
            y={460}
          />
          <Rect
            height={18}
            radius={9}
            width={420}
            y={500}
          />
        </ContentLoader>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Rect
              height={400}
              width={680}
            />,
            <Rect
              height={18}
              radius={9}
              width={680}
              y={420}
            />,
            <Rect
              height={18}
              radius={9}
              width={500}
              y={460}
            />,
            <Rect
              height={18}
              radius={9}
              width={420}
              y={500}
            />,
          ],
          "height": 700,
          "primaryColor": "#f5d2bd",
          "secondaryColor": "#f9e0d2",
          "speed": 1,
          "type": "facebook",
          "width": 680,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "height": 400,
              "width": 680,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "height": 18,
              "radius": 9,
              "width": 680,
              "y": 420,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "height": 18,
              "radius": 9,
              "width": 500,
              "y": 460,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "height": 18,
              "radius": 9,
              "width": 420,
              "y": 500,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
